vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py

```diff
@@ -1296,7 +1333,20 @@ def _pop_done_transfers(
                     in_progress = True
                     continue
                 else:
-                    raise RuntimeError("Transfer failed with state %s", xfer_state)
+                    # transfer failed - mark blocks as invalid
+                    logger.error(
+                        "NIXL transfer failed for request %s with state %s. "
+                        "Marking blocks as invalid.",
+                        req_id,
+                        xfer_state,
+                    )
+                    # mark all blocks for this request as invalid
+                    meta = self._recving_metadata.get(req_id)
+                    if meta:
+                        self._invalid_block_ids.update(meta.local_block_ids)
+                    self._recving_metadata.pop(req_id, None)
```

Comment:
suggest simplification:
```suggestion
                    if meta := self._recving_metadata.pop(req_id, None):
                        self._invalid_block_ids.update(meta.local_block_ids)
```